modules/templates/templates_unattended_api.rb

```diff
@@ -13,19 +13,19 @@ class Proxy::TemplatesUnattendedApi < Sinatra::Base
   end
 
   get "/:kind/:template/:hostgroup" do |kind, template, hostgroup|
-    log_halt(500, "Failed to retrieve #{kind} hostgroup template for #{params.inspect}: ") do
+    log_halt(nil, "Failed to retrieve #{kind} hostgroup template for #{params.inspect}: ") do
```

Comment:
If you don't want it to halt, then don't use `log_halt`. Instead, implement your own exception handling.

Reply:
I want to use log_halt, I just want to set my own status code, checking the `log_halt` `code` is optional:
https://github.com/theforeman/smart-proxy/blob/70073c07e991cbe85a332a00e90d4196d8cbcaaf/lib/proxy/helpers.rb#L10